### DISCUSSION TITLE
Dot notation in v3

### DISCUSSION THREAD

**Discussion by SasSam** [chakra-ui/chakra-ui#8834] [General]
Hi,

I've just checked the v3 documentation, and I spotted a section in the migration guide. It says:

> "Namespaced imports: Import components using the dot notation for more concise imports"

I'm really a fan of dot notation in components, but RSC (React 19) seems to cause unresolvable issues. At least many other libraries/packages suffer from this issue!

If you guys have not seen this issue before, then it is worth checking this ASAP!

References:
- https://github.com/vercel/next.js/issues/51593
- https://github.com/react-bootstrap/react-bootstrap/issues/6669

**Comment by segunadebayo** [comment]
Hey @SasSam. Yes, we're aware of this.

It all depends on how the dot notation was achieved. If you use `Object.assign()` or something like `const Accordion = { ... }`, it won't work.

However, if you use the namespace imports `export * as Accordion`, it works just fine.

**Reply by SasSam** [reply]
Awesome news, thanks!
